### PR TITLE
Feat: ctrl-plane command to start vela control plane in one step

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: qiaozp <chivalry.pp@gmail.com>

### Description of your changes

This PR mainly do two things

1. provide a CLI command `vela ctrl-plane`, which automatically install k3s and vela-core in it. Because of the new multicluster pattern used in KubeVela (mainly based on `ref-object` type component). This command will install k3s without workload controllers (Deployment, Job, etc) by defualt. This conmmand only work in Linux for now.

```shell
$ vela ctrl-plane -h
Quickly setup a KubeVela control plane, using K3s and only for Linux now

Usage:
  vela ctrl-plane [command]

Available Commands:
  install     Quickly setup a KubeVela control plane
  kubeconfig  print kubeconfig to access control plane
  uninstall   uninstall control plane

Flags:
  -h, --help   help for ctrl-plane

Global Flags:
  -y, --yes   Assume yes for all user prompts

Use "vela ctrl-plane [command] --help" for more information about a command.
```

2. Now our Helm chart contains several Deployment/Job manifests. We want to use vela-core's multicluster in control plane standalone mode, which means install the chart in a cluster whose controller-manager is incomplete. This PR provide a parameter: podOnly in vela-core chart. It won't affect the default install behavior. When set podOnly=true. All Deployments will be replaced with Pod with the same spec. Here is the replaced manifest list :

  - kubevela-controller
  - cluster-gateway
  - kubevela-vela-core-admission-create
  - kubevela-vela-core-admission-patch

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by Somefive — APPROVED** [review]

**Review by wonderflow — CHANGES_REQUESTED** [review]
This change will make the chart hardly to maintain. Is there any alternative solution?

I think we can create a pod spec yaml and use the same values.yaml?

**Comment by chivalryq** [comment]
>

**Comment by chivalryq** [comment]
> This change will make the chart hardly to maintain. Is there any alternative solution?

> I think we can create a pod spec yaml and use the same values.yaml?

Do you mean helm chart is like below?

```
#deploy.yaml
{{ if not podOnly }}
Deployment
{{ end }}

# pod.yaml
{{ if .Values.PodOnly }}
Pod
{{ end }}
```

This creates a duplicate copy.

**Comment by chivalryq** [comment]
ping @wonderflow please review again.

**Comment by wonderflow** [comment]
let's move this feature into a seprate project https://github.com/oam-dev/velad

File: charts/vela-core/values.yaml [hunk 1]
```diff
@@ -210,3 +210,6 @@ admissionWebhooks:
 kubeClient:
   qps: 50
   burst: 100
+
+## @param podOnly Use pod instead of Deployment to deploy vela-core
+podOnly: false
```
Comment (wonderflow) [inline charts/vela-core/values.yaml:215]:
```suggestion
deployByPod: false
```

**Pull request closed without merging**